### ISSUE TITLE
init_params in BoxOSQP().run

### ISSUE THREAD

**Issue by HajimeKawahara** [google/jaxopt#181]
Thanks for a very cool and useful library! I have a question on `inti_params` in BoxOSQP().run.
I just copy and run the code In [the tutorial of BoxOSQP.](https://jaxopt.github.io/stable/quadratic_programming.html#box-constrained-qps):

```python
import jax.numpy as jnp
from jaxopt import BoxOSQP

Q = 2 * jnp.array([[2.0, 0.5], [0.5, 1]])
c = jnp.array([1.0, 1.0])
A = jnp.array([[1.0, 1.0], [-1.0, 0.0], [0.0, -1.0]])
l = jnp.array([1.0, -jnp.inf, -jnp.inf])
u = jnp.array([1.0, 0.0, 0.0])

qp = BoxOSQP()
sol = qp.run(params_obj=(Q, c), params_eq=A, params_ineq=(l, u)).params
```
and got
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_11430/875702596.py in <module>
      9 
     10 qp = BoxOSQP()
---> 11 sol = qp.run(params_obj=(Q, c), params_eq=A, params_ineq=(l, u)).params
     12 
     13 print(sol.primal)

TypeError: run() missing 1 required positional argument: 'init_params'
```
So, just adding `init_params=None` in the optiion of run,  it worked. But I am not confident that it is OK or not.

```python
sol = qp.run(init_params=None,params_obj=(Q, c), params_eq=A, params_ineq=(l, u)).params

print(sol.primal)
print(sol.dual_eq)
print(sol.dual_ineq)

(DeviceArray([0.25004143, 0.7500388 ], dtype=float32), DeviceArray([ 1.        , -0.25003824, -0.75000846], dtype=float32))
[-2.7502573e+00  0.0000000e+00  3.0822962e-09]
(DeviceArray([0.0000000e+00, 0.0000000e+00, 3.0822962e-09], dtype=float32), DeviceArray([ 2.7502573, -0.       ,  0.       ], dtype=float32))
```

Thanks in advance.

**Comment by HajimeKawahara** [comment]
Note that I use jaxopt=0.3.

**Comment by Algue-Rythme** [comment]
Hi HajimeKawahara
This will be fixed in a new version of Jaxopt and `init_params` will no longer be required.
It is Ok to do what you did and it is the intended behavior: when the initial KKT solution is unspecified BoxOSQP choses one for you (with x=0). Other KKT solutions may be useful with warm start, for example if you solve a sequence of similar problems. Let me know if you have any question !

**Comment by mblondel** [comment]
I just released version 0.3.1 which should now set `init_params=None` by default. Please run `pip install jaxopt` :) Please reopen if the issue is still present.

**Comment by HajimeKawahara** [comment]
I confirmed It worked w/o explicit init_params after `pip install -U jaxopt`. Thanks!
